### ISSUE TITLE
[BUG] Credential persistence did not work in Mac M1

### ISSUE THREAD

**Issue by Flanker32** [Azure/azure-sdk-for-java#29308]
**Describe the bug**
`TokenCredentialBuilder` (OAuth/Device Code) with tokenCachePersistenceOptions enabled did not work in Mac M1. Which seems to be an issue of the old JNA library, refers https://github.com/java-native-access/jna/pull/1238

***Exception or Stack Trace***
```
[ForkJoinPool.commonPool-worker-1] ERROR com.azure.identity.InteractiveBrowserCredential - Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: Failed to acquire token with Interactive Browser Authentication.
[ForkJoinPool.commonPool-worker-1] ERROR com.azure.identity.InteractiveBrowserCredential - Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: Failed to acquire token with Interactive Browser Authentication.
[ForkJoinPool.commonPool-worker-2] ERROR com.azure.identity.InteractiveBrowserCredential - Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: Failed to acquire token with Interactive Browser Authentication.
[ForkJoinPool.commonPool-worker-1] ERROR com.azure.identity.InteractiveBrowserCredential - Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: Failed to acquire token with Interactive Browser Authentication.
Exception in thread "main" com.azure.core.exception.ClientAuthenticationException: Failed to acquire token with Interactive Browser Authentication.
	at com.azure.identity.implementation.IdentityClient.lambda$authenticateWithBrowserInteraction$46(IdentityClient.java:971)
	at reactor.core.publisher.Mono.lambda$onErrorMap$31(Mono.java:3733)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
	at reactor.core.publisher.MonoCompletionStage.lambda$subscribe$0(MonoCompletionStage.java:77)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingOptionalMonoSubscriber.blockingGet(BlockingOptionalMonoSubscriber.java:123)
		at reactor.core.publisher.Mono.blockOptional(Mono.java:1752)
		at com.azure.identity.test.AzureIdentityExample.printToken(AzureIdentityExample.java:53)
		at com.azure.identity.test.AzureIdentityExample.main(AzureIdentityExample.java:40)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.microsoft.aad.msal4jextensions.persistence.mac.ISecurityLibrary
	at com.microsoft.aad.msal4jextensions.persistence.mac.KeyChainAccessor.read(KeyChainAccessor.java:49)
	at com.microsoft.aad.msal4jextensions.PersistenceTokenCacheAccessAspect.beforeCacheAccess(PersistenceTokenCacheAccessAspect.java:125)
	at com.azure.identity.implementation.PersistentTokenCacheImpl.beforeCacheAccess(PersistentTokenCacheImpl.java:68)
	at com.microsoft.aad.msal4j.TokenCache$CacheAspect.<init>(TokenCache.java:164)
	at com.microsoft.aad.msal4j.TokenCache.saveTokens(TokenCache.java:181)
	at com.microsoft.aad.msal4j.AbstractClientApplicationBase.acquireTokenCommon(AbstractClientApplicationBase.java:131)
	at com.microsoft.aad.msal4j.AcquireTokenByAuthorizationGrantSupplier.execute(AcquireTokenByAuthorizationGrantSupplier.java:63)
	at com.microsoft.aad.msal4j.AcquireTokenByInteractiveFlowSupplier.acquireTokenWithAuthorizationCode(AcquireTokenByInteractiveFlowSupplier.java:185)
	at com.microsoft.aad.msal4j.AcquireTokenByInteractiveFlowSupplier.execute(AcquireTokenByInteractiveFlowSupplier.java:39)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:71)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:20)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 more
```

**To Reproduce**
Create a token credential with token cache persist, may refer the code snippet below

***Code Snippet***
Add the code snippet that causes the issue.

```java
    public static void main(String[] args) {
        TokenCredential credential = signinWithOAuth();
        printToken(credential);
        AzureProfile azureProfile = new AzureProfile(tenantId, subscriptionId, environment);
        final AppServiceManager manager = AppServiceManager.configure()
            .withLogLevel(HttpLogDetailLevel.HEADERS)
            .authenticate(credential, azureProfile);
        manager.webApps().list().stream().map(HasName::name).sorted(Comparator.comparing(s -> s)).forEach(System.out::println);
    }

    private static TokenCredential signinWithOAuth() {
        return new InteractiveBrowserCredentialBuilder()
            .clientId(clientId)
            .tokenCachePersistenceOptions(new TokenCachePersistenceOptions().setName(tokenCacheName))
            .redirectUrl("http://localhost:1101")
            .build();
    }
```

**Expected behavior**
Credential persistence should works in M1 Mac. After investigation, it should be an issue of the JNA library. Identity will call `com.microsoft.aad.msal4jextensions.persistence.mac.ISecurityLibrary` for security storage in Mac, which will leverage JNA `Native`, however, version of JNA shipped with identity is 5.5.0, which did not support Mac M1, refers https://github.com/java-native-access/jna/pull/1238, after import the latest JNA 5.11.0, the issue was resolved and we could sign in and restore sign in successfully with identity.

```
[INFO] +- com.azure:azure-identity:jar:1.5.1:compile
[INFO] |  +- com.microsoft.azure:msal4j:jar:1.12.0:compile
[INFO] |  |  +- com.nimbusds:oauth2-oidc-sdk:jar:9.32:compile
[INFO] |  |  |  +- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:compile
[INFO] |  |  |  +- com.nimbusds:content-type:jar:2.2:compile
[INFO] |  |  |  +- com.nimbusds:lang-tag:jar:1.6:compile
[INFO] |  |  |  \- com.nimbusds:nimbus-jose-jwt:jar:9.21:compile
[INFO] |  |  \- net.minidev:json-smart:jar:2.4.8:compile
[INFO] |  |     \- net.minidev:accessors-smart:jar:2.4.8:compile
[INFO] |  |        \- org.ow2.asm:asm:jar:9.1:compile
[INFO] |  +- com.microsoft.azure:msal4j-persistence-extension:jar:1.1.0:compile
[INFO] |  |  \- net.java.dev.jna:jna:jar:5.5.0:compile
[INFO] |  \- net.java.dev.jna:jna-platform:jar:5.6.0:compile

```

**Screenshots**

**Setup (please complete the following information):**
 - OS: Mac Monterey (M1)
 - IDE: IntelliJ
 - Library/Libraries: com.azure:azure-sdk-bom:1.2.2
 - Java version: Azul 17.0.3

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
Thanks for reporting this issue and explaining the root cause as well as the fix, @Flanker32.  @billwert could you please follow up on the msal dependency issue and link back to this issue, as appropriate?

/cc @g2vinay

**Comment by mitchross** [comment]
When will this be fixed?

**Comment by billwert** [comment]
@siddhijain - this is a bug in `msal4j-persistence-extension`. I'm not sure if you own this or not - it hasn't been touched in a few years. Can you take a look?

**Comment by siddhijain** [comment]
@billwert I will take a look. Do you know if there is already a github issue logged in msal persistence layer for this?

**Comment by billwert** [comment]
@siddhijain I don't believe so. I didn't want to move it until you'd taken a peek to see if you agreed that's where the issue was. Thanks!

**Comment by siddhijain** [comment]
I released version 1.13.2 of msal-persistence-extensions to take care of this issue. Let me know if the issue is resolved.

**Comment by scottaddie** [comment]
Minor correction to the above. v1.2.0 is the version that fixes this issue.
